drivers/hmis/app/models/hmis/activity_log.rb

```diff
@@ -4,7 +4,25 @@
 # License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
 ###
 
+# = Hmis::ActivityLog
+#
+# Tracks GraphQL API access
 class Hmis::ActivityLog < ApplicationRecord
   self.table_name = :hmis_activity_logs
   belongs_to :user, class_name: 'Hmis::User'
+  belongs_to :data_source, class_name: 'GrdaWarehouse::DataSource'
+
+  scope :unprocessed, -> { where(processed_at: nil) }
+
+  def self.select_client_ids
+    jt = Arel::Table.new(:hmis_activity_logs_clients)
+    join_clause = arel_table.create_join(jt, arel_table.create_on(jt[:activity_log_id].eq(arel_table[:id])))
+    joins(join_clause).select(jt[:client_id])
+  end
+
+  def self.select_enrollment_ids
+    jt = Arel::Table.new(:hmis_activity_logs_enrollments)
+    join_clause = arel_table.create_join(jt, arel_table.create_on(jt[:activity_log_id].eq(arel_table[:id])))
+    joins(join_clause).select(jt[:enrollment_id])
+  end
```

Comment:
This also feels like we're rebuilding some active record.  Is there a reason we're avoiding using relations?  If so, let'd document why.

Reply:
I'll add some documentation. These join tables span the app and warehouse database boundary, that's why I avoided using active record associations here. Active record associations seem like they would be hard to use without accidentally creating a cross-db join.

Reply:
Are you getting the cross-db join to actually work?!?  That'd be great.  Active record hasn't shown an ability to do that I'm aware of, but we have a handful of places it would be really useful.  We do have relations that cross DB boundaries, but you can't do things like `User.joins(:clients)` because of the cross-db nature of that.  Our approach, historically, has been to pluck IDs and force two queries.

Reply:
No, sorry I didn't mean to imply I got cross-db joins working. I didn't try it, although supposedly it is possible. I fetch the ids down as you describe to bridge the boundary.

I did see Rails 7 supports `disable_joins` on associations which seems to take care of the fetch/stitch operation for you automatically